src/engraving/dom/edit.cpp

```diff
@@ -2993,6 +2993,21 @@ void Score::deleteMeasures(MeasureBase* mbStart, MeasureBase* mbEnd, bool preser
         Measure* startMeasure = score->tick2measure(startTick);
         Measure* endMeasure = score->tick2measure(endTick);
 
+        // delete staffTypeChanges in removed measures
+        for (Measure* m = startMeasure; m != endMeasure->nextMeasure() && m != nullptr; m = m->nextMeasure()) {
+            for (Staff* staff : score->staves()) {
+                if (staff->isStaffTypeStartFrom(m->tick())) {
+                    for (size_t i = m->el().size(); i > 0; --i) {
+                        EngravingItem* el = m->el().at(i - 1);
+                        if (el && el->isStaffTypeChange()) {
+                            score->deleteItem(el);
+                        }
+                    }
+                    break;
```

Comment:
Shouldn't this be two lines above instead? (Now, it won't check other staves after finding a staff type change on one staff; but perhaps you meant to stop checking other measure elements after finding the staff type change)

Reply:
@sammik Sorry, I'm afraid I don't entirely understand what you mean; but does it look okay now?
(I pushed 1 commit containing fixes for these two review comments, and also did a rebase (but for the rest didn't touch your commits)